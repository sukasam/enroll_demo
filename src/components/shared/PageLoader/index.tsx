import Box from "@mui/material/Box";
import CircularProgress from "@mui/material/CircularProgress";

function PageLoader(): React.ReactElement {
    return (
        <Box
            sx={{
                position: "absolute",
                top: 0,
                left: 0,
                width: "100%",
                height: "100%",
                display: "flex",
                alignItems: "center",
                justifyContent: "center",
                backgroundColor: "rgba(255, 255, 255, 0.5)",
                zIndex: 1500
            }}
        >
            <CircularProgress sx={{ color: "#5a8fc4" }} thickness={5} />
        </Box>
    );
}

export default PageLoader;
