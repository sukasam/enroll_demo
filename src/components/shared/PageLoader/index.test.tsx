import { render } from "@testing-library/react";
import PageLoader from "./index";

jest.mock(
    "@mui/material/Box",
    () =>
        function MockBox(props: any): JSX.Element {
            return <div data-testid="mock-box" {...props} />;
        }
);
jest.mock(
    "@mui/material/CircularProgress",
    () =>
        function MockCircularProgress(props: any): JSX.Element {
            return <div data-testid="mock-circular-progress" {...props} />;
        }
);

describe("PageLoader", (): void => {
    it("matches snapshot", (): void => {
        const { container } = render(<PageLoader />);
        expect(container).toMatchSnapshot();
    });
});
