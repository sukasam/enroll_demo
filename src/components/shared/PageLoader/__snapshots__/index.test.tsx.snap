// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoader matches snapshot 1`] = `
<div>
  <div
    data-testid="mock-box"
    sx="[object Object]"
  >
    <div
      data-testid="mock-circular-progress"
      sx="[object Object]"
      thickness="5"
    />
  </div>
</div>
`;
